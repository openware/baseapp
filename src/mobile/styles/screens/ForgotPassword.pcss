/* stylelint-disable */
.pg-mobile-app {
  .pg-forgot-password-screen {
    background: unset;
    margin: 0 14px;
    .pg-forgot-password-screen__container {
      margin: 0;
    }
    .pg-forgot-password___form {
      .cr-email-form__header {
        font-size: 14px;
        color: var(--primary-text-color);
        margin: 24px 0 0 0;
        width: auto;
      }
    }
  }
  .cr-email-form {
    background: unset;
    margin: 0 auto;
  }
  .cr-email-form__form-content {
    position: static;
  }
  .cr-email-form__button-wrapper {
    position: static;
    button {
      height: 41px;
      text-transform: uppercase;
      font-size: 14px;
    }
  }

  .custom-input {
    font-size: 12px;

    input {
      font-size: 14px;
      &::placeholder {
        font-size: 14px;
      }
    }
  }

  .cr-email-form__group--focused input:focus {
    border-width: 2px;
    border-color: var(--button-primary-cta-background-color);
  }
}
