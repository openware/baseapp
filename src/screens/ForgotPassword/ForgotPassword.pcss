.pg-forgot-password-screen {
  background: url(assets/images/background.svg);
  background-size: cover;
  display: flex;
  height: calc(100vh - 95px);
  justify-content: center;
  margin: 0 auto;

  .pg-forgot-password-screen__container {
    background: transparent;
    margin-top: calc(var(--gap) * 3.33);
    width: calc(var(--gap) * 70.66);
  }

  .pg-forgot-password___form {
    .cr-email-form {
      background: var(--body-background-color);
      min-height: calc(var(--big-gap) * 5);

      &__option {
        background: var(--subheader-background-color);
      }

      &__options-group {
        border-bottom: 2px solid var(--divider-color-level-2);
        height: calc(var(--big-gap) * 0.97);
        width: 100%;
      }

      &__label {
        background: var(--input-background-color);
        position: absolute;
        top: 0;
      }

      &__input {
        input {
          border: 1px solid var(--input-border-color);
          border-radius: calc(var(--gap) * 0.66);
          color: var(--input-text-color);
          font-size: calc(var(--gap) * 2.33);
        }
      }

      &__header {
        color: var(--secondary-contrast-cta-color);
        font-size: calc(var(--gap) * 2.33);
        font-style: normal;
        font-weight: normal;
        height: calc(var(--gap) * 4.33);
        line-height: calc(var(--gap) * 2.66);
        margin: calc(var(--gap) * 0.83);
        width: calc(var(--big-gap) * 6.3);
      }

      &__cros-icon {
        cursor: pointer;
        display: inline-block;
        height: calc(var(--gap) * 2.5);
        margin-top: calc(var(--gap) * 0.5);
        position: absolute;
        right: calc(var(--gap) * 3.3);
        width: calc(var(--gap) * 2.5);
      }

      &__group--focused {
        .cr-sign-in-form__label {
          color: var(--input-legend-color-active);
        }

        input {
          border-color: var(--input-border-color-active) !important;
        }
      }

      &__button-wrapper {
        min-height: calc(var(--gap) * 10);
      }
    }
  }
}

@media screen and (max-width: 768px) {
  .pg-forgot-password-screen {
    .pg-forgot-password-screen__container {
      margin: calc(var(--gap) * 3.33) calc(var(--gap) * 1.66) 0;
      width: 100%;
    }
  }
}
