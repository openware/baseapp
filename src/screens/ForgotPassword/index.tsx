import * as React from 'react';
import {
    injectIntl,
} from 'react-intl';
import {
    connect,
    MapDispatchToPropsFunction,
    MapStateToProps,
} from 'react-redux';
import { RouterProps } from 'react-router';
import { withRouter } from 'react-router-dom';
import { compose } from 'redux';
import { IntlProps } from '../../';
import { Captcha, EmailForm } from '../../components';
import {
    EMAIL_REGEX,
    ERROR_INVALID_EMAIL,
    setDocumentTitle,
} from '../../helpers';
import {
    Configs,
    forgotPassword,
    GeetestCaptchaResponse,
    resetCaptchaState,
    RootState,
    selectCaptchaResponse,
    selectConfigs,
    selectCurrentLanguage,
    selectForgotPasswordError,
    selectForgotPasswordSuccess,
    selectGeetestCaptchaSuccess,
    selectRecaptchaSuccess,
} from '../../modules';
import { CommonError } from '../../modules/types';

interface ReduxProps {
    success: boolean;
    error?: CommonError;
    configs: Configs;
    captcha_response?: string | GeetestCaptchaResponse;
    reCaptchaSuccess: boolean;
    geetestCaptchaSuccess: boolean;
}

interface DispatchProps {
    forgotPassword: typeof forgotPassword;
    resetCaptchaState: typeof resetCaptchaState;
}

interface ForgotPasswordState {
    email: string;
    emailError: string;
    emailFocused: boolean;
}

type Props = RouterProps & ReduxProps & DispatchProps & IntlProps;

class ForgotPasswordComponent extends React.Component<Props, ForgotPasswordState> {
    constructor(props: Props) {
        super(props);

        this.state = {
            email: '',
            emailError: '',
            emailFocused: false,
        };
    }

    public componentDidMount() {
        setDocumentTitle('Forgot password');
    }

    public renderCaptcha = () => {
        const { error, success } = this.props;

        return (
            <Captcha
                error={error}
                success={success}
            />
        );
    };

    public render() {
        const {
            email,
            emailFocused,
            emailError,
        } = this.state;
        const {
            configs,
            captcha_response,
            reCaptchaSuccess,
            geetestCaptchaSuccess,
        } = this.props;

        return (
            <div className="pg-forgot-password-screen" onKeyPress={this.handleEnterPress}>
                <div className="pg-forgot-password-screen__container">
                    <div className="pg-forgot-password___form">
                        <EmailForm
                            OnSubmit={this.handleChangePassword}
                            title={this.props.intl.formatMessage({id: 'page.forgotPassword'})}
                            emailLabel={this.props.intl.formatMessage({id: 'page.forgotPassword.email'})}
                            buttonLabel={this.props.intl.formatMessage({id: 'page.forgotPassword.send'})}
                            email={email}
                            emailFocused={emailFocused}
                            emailError={emailError}
                            message={this.props.intl.formatMessage({id: 'page.forgotPassword.message'})}
                            validateForm={this.validateForm}
                            handleInputEmail={this.handleInputEmail}
                            handleFieldFocus={this.handleFocusEmail}
                            handleReturnBack={this.handleComeBack}
                            captchaType={configs.captcha_type}
                            renderCaptcha={this.renderCaptcha()}
                            reCaptchaSuccess={reCaptchaSuccess}
                            geetestCaptchaSuccess={geetestCaptchaSuccess}
                            captcha_response={captcha_response}
                        />
                    </div>
                </div>
            </div>
        );
    }

    private handleChangePassword = () => {
        const { email } = this.state;
        const { configs, captcha_response } = this.props;

        switch (configs.captcha_type) {
            case 'recaptcha':
            case 'geetest':
                this.props.forgotPassword({ email, captcha_response });
                break;
            default:
                this.props.forgotPassword({ email });
                break;
        }

        this.setState({ email: '' });

        this.props.resetCaptchaState();
    };

    private handleFocusEmail = () => {
        this.setState({
            emailFocused: !this.state.emailFocused,
        });
    };

    private handleInputEmail = (value: string) => {
        this.setState({
            email: value,
        });
    };

    private validateForm = () => {
        const { email } = this.state;

        const isEmailValid = email ? email.match(EMAIL_REGEX) : true;

        if (!isEmailValid) {
            this.setState({
                emailError: ERROR_INVALID_EMAIL,
            });

            return;
        }
    };

    private handleComeBack = () => {
        this.props.history.goBack();
    };

    private handleEnterPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
        if (event.key === 'Enter') {
            event.preventDefault();

            this.handleChangePassword();
        }
    };
}

const mapStateToProps: MapStateToProps<ReduxProps, {}, RootState> = state => ({
    success: selectForgotPasswordSuccess(state),
    error: selectForgotPasswordError(state),
    i18n: selectCurrentLanguage(state),
    configs: selectConfigs(state),
    captcha_response: selectCaptchaResponse(state),
    reCaptchaSuccess: selectRecaptchaSuccess(state),
    geetestCaptchaSuccess: selectGeetestCaptchaSuccess(state),
});

const mapDispatchToProps: MapDispatchToPropsFunction<DispatchProps, {}> =
    dispatch => ({
        forgotPassword: credentials => dispatch(forgotPassword(credentials)),
        resetCaptchaState: () => dispatch(resetCaptchaState()),
    });

export const ForgotPasswordScreen = compose(
    injectIntl,
    withRouter,
    connect(mapStateToProps, mapDispatchToProps),
)(ForgotPasswordComponent) as React.ComponentClass;
