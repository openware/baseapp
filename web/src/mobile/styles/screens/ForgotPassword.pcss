.pg-mobile-app {
  .pg-forgot-password-screen {
    background: unset;
    height: max-content;
    margin: 0 14px;

    .pg-forgot-password-screen__container {
      margin: 0;
    }

    .pg-forgot-password___form .cr-email-form {
      min-height: unset;

      .cr-email-form__header {
        color: var(--primary-text-color);
        font-size: 14px;
        margin: 24px 0 0 0;
        width: auto;
      }
    }
  }
}
