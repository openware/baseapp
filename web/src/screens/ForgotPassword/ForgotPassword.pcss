.pg-forgot-password-screen {
  background: url(assets/images/background.svg);
  background-size: cover;
  display: flex;
  height: calc(100vh - 95px);
  justify-content: center;
  margin: 0 auto;

  .pg-forgot-password-screen__container {
    background: transparent;
    margin-top: calc(var(--gap) * 3.33);
    width: calc(var(--gap) * 70.66);
  }

  .pg-forgot-password___form {
    .cr-email-form {
      background: var(--body-background-color);
      min-height: calc(var(--big-gap) * 3.9);

      &__option {
        background: var(--subheader-background-color);
      }

      &__options-group {
        border-bottom: 2px solid var(--divider-color-level-2);
        height: calc(var(--big-gap) * 0.97);
        width: 100%;
      }

      &__label {
        background: none;
        font-size: 10px;
        font-weight: bold;
        margin: 0;
        position: static;
        padding: 0;
        text-transform: uppercase;
      }

      .input-group-lg {
        margin-top: 4px;

        .form-control {
          border: 2px solid transparent;
          height: 48px;
        }
      }

      .cr-email-form__group--focused {
        input {
          border: 2px solid var(--input-border-color-active) !important;
        }
      }

      &__header {
        color: var(--secondary-contrast-cta-color);
        font-size: calc(var(--gap) * 2.33);
        font-style: normal;
        font-weight: normal;
        line-height: calc(var(--gap) * 2.66);
        margin-bottom: calc(var(--gap) * 0.83);
        width: calc(var(--big-gap) * 6.3);
      }

      &__cros-icon {
        cursor: pointer;
        display: inline-block;
        height: calc(var(--gap) * 2.5);
        position: absolute;
        right: calc(var(--gap) * 3.3);
        width: calc(var(--gap) * 2.5);
      }

      &__group--focused {
        .cr-sign-in-form__label {
          color: var(--input-legend-color-active);
        }

        input {
          border-color: var(--input-border-color-active) !important;
        }
      }

      &__button-wrapper {
        min-height: max-content;

        button {
          min-height: 40px;
        }
      }
    }
  }
}

@media screen and (max-width: 768px) {
  .pg-forgot-password-screen {
    .pg-forgot-password-screen__container {
      margin: calc(var(--gap) * 3.33) calc(var(--gap) * 1.66) 0;
      width: 100%;
    }
  }
}
