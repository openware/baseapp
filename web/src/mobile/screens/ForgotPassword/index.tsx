import * as React from 'react';
import { useIntl } from 'react-intl';
import { useHistory } from 'react-router';
import { ForgotPasswordScreen } from '../../../screens/ForgotPassword';
import { Modal } from '../../components';

const ForgotPasswordMobileScreen: React.FC = () => {
    const history = useHistory();
    const intl = useIntl();

    return (
        <div>
            <Modal
                isOpen={true}
                onClose={() => history.push('/trading')}
                onBack={() => history.push('/signin')}
                backTitle={intl.formatMessage({ id: 'page.body.landing.header.button2' })}
                title={intl.formatMessage({ id: 'page.forgotPassword' })}>
                <ForgotPasswordScreen/>
            </Modal>
        </div>
    );
};

export {
    ForgotPasswordMobileScreen,
};
