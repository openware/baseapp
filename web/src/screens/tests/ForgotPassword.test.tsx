import { shallow } from 'enzyme';
import React from 'react';
import { TestComponentWrapper } from 'lib/test';
import { IntlProps } from '../../';
import { ForgotPasswordScreen } from '../ForgotPassword';

const setup = (props: Partial<IntlProps> = {}) =>
    shallow(
        <TestComponentWrapper>
            <ForgotPasswordScreen />
        </TestComponentWrapper>
    );

describe('ForgotPasswordScreen', () => {
    it('should render', () => {
        const wrapper = setup().render();
        expect(wrapper).toMatchSnapshot();
    });
});
