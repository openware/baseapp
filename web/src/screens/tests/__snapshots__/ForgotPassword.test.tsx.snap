// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPasswordScreen should render 1`] = `
<div
  class="pg-forgot-password-screen"
>
  <div
    class="pg-forgot-password-screen__container"
  >
    <div
      class="pg-forgot-password___form"
    >
      <form>
        <div
          class="cr-email-form"
        >
          <div
            class="cr-email-form__options-group"
          >
            <div
              class="cr-email-form__option"
            >
              <div
                class="cr-email-form__option-inner"
              >
                Forgot Password
                <div
                  class="cr-email-form__cros-icon"
                >
                  <svg
                    class="close-icon"
                    fill="none"
                    height="14"
                    viewBox="0 0 14 14"
                    width="14"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M5.58434 6.99901L0.875873 11.7079L2.29003 13.1221L6.99849 8.41328L11.707 13.1221L13.1211 11.7079L8.41265 6.99901L13.704 1.7072L12.2899 0.292927L6.99849 5.58474L1.70712 0.292932L0.292969 1.7072L5.58434 6.99901Z"
                      fill="var(--icons)"
                      fill-rule="evenodd"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <div
            class="cr-email-form__form-content"
          >
            <div
              class="cr-email-form__header"
            >
              Enter email to reset password
            </div>
            <div
              class="cr-email-form__group"
            >
              <div
                class="custom-input"
              >
                <label
                  class="cr-email-form__label"
                />
                <div
                  class="input-group input-group-lg"
                >
                  <input
                    autofocus=""
                    class="form-control form-control-lg"
                    placeholder="Email"
                    type="email"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              class="cr-email-form__button-wrapper"
            >
              <button
                class="btn btn-primary btn-block btn-lg"
                disabled=""
                type="button"
              >
                Send
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
